File: src/Desafio.Domain/Entidades/Proprietario.cs
using Desafio.Domain.Enums;
using Desafio.Domain.ValueObject;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Desafio.Domain.Entidades
{
    public class Proprietario : Entity
    {
        public Proprietario(Nome nome, int documento, string email, Endereco endereco, EStatus status)
        {
            Nome = nome;
            Documento = documento;
            Email = email;
            Endereco = endereco;
            _veiculos = new List<Veiculo>();
            Status = status;
        }
        protected Proprietario()
        {

        }

        public Nome Nome { get; private set; }
        public int Documento { get; private set; }
        public string Email { get; private set; }
        public Endereco Endereco { get; private set; }
        public EStatus Status { get; private set; }
        private readonly List<Veiculo> _veiculos;
        public IReadOnlyCollection<Veiculo> Veiculos => _veiculos;

        public void Alterar(Nome nome, int documento, string email, Endereco endereco)
        {
            Nome = nome;
            Documento = documento;
            Email = email;
            Endereco = endereco;
        }

        public void Ativar()
        {
            Status = EStatus.Ativo;
        }

        public void Cancelar()
        {
            Status = EStatus.Cancelado;
        }

    }
}

File: Desafio-FrontEnd/src/app/models/proprietario.ts
import { Endereco } from "./endereco";
import { Nome } from "./nome";

export class Proprietario {

    constructor(
        id?: string,
        nome?: Nome,
        email?: string,
        endereco?: Endereco,
        status?: number,
        documento?: number) {
            this.id = id;
            this.nome = nome;
            this.email = email;
            this.endereco = endereco;
            this.status = status;
            this.documento = documento;
    }

    public id?: string;
    public nome?: Nome;
    public email?: string;
    public endereco?: Endereco;
    public status?: number;
    public documento?: number
}